### DISCUSSION TITLE
Python tuple to array conversion

### DISCUSSION THREAD

**Discussion by eaubin** [duckdb/duckdb#5741] [Ideas]
DuckDB seems to like python lists a tiny bit better than tuples.  E.g. you can insert from a dataframe containing a list series without doing anything special:

```python 
con = duckdb.connect()
con.execute("CREATE OR REPLACE TABLE test (nums INTEGER[])")
list_df = pd.DataFrame.from_dict(dict(nums=[[1,2,3],[4,5,6]]))
con.execute("INSERT INTO test (nums) SELECT * FROM list_df");
con.execute("SELECT * FROM test").df()
```

but inserting from a tuple like:

```python
tuple_df = pd.DataFrame.from_dict(dict(nums=[(1,2,3,),(4,5,6,)]))
con.execute("INSERT INTO test (nums) SELECT * FROM tuple_df ");
```

gives a conversion error `Conversion Error: Type VARCHAR with value '(1, 2, 3)' can't be cast to the destination type LIST`.

I like tuples because they're hashable and easy to use in a dictionary or set, so I use them and convert the dataframe contents back to lists before INSERTing.  Would it make sense for the python client to convert tuples to arrays like it does for lists?  Is there a way to avoid the extra conversion back to lists?

**Comment by Tishj** [comment]
Hey, I completely missed this at the time, it seems we did not check for the tuple type, and it instead got handled as "other" which is why it was cast to VARCHAR instead

**Reply by Tishj** [reply]
Should be fixed by #6868
